rust/src/rfb/rfb.rs

```diff
@@ -308,20 +342,25 @@ impl RFBState {
                         );
                     }
                     Err(_) => {
-                        return AppLayerResult::err();
+                        if let Some(current_transaction) = self.get_current_tx() {
+                            RFBState::set_event(current_transaction, RFBEvent::MalformedMessage);
+                        }
+                        // We failed to parse the client init.
+                        // Continue the flow but stop trying to map the protocol.
+                        self.state = parser::RFBGlobalState::Skip;
+                        return AppLayerResult::ok();
                     }
                 },
-                parser::RFBGlobalState::Message => {
-                    //todo implement RFB messages, for now we stop here
-                    return AppLayerResult::err();
-                }
-                parser::RFBGlobalState::TCServerProtocolVersion => {
-                    SCLogDebug!("Reversed traffic, expected response.");
-                    return AppLayerResult::err();
+                parser::RFBGlobalState::Skip => {
+                    // End of parseable handshake reached, skip rest of traffic
+                    return AppLayerResult::ok();
                 }
                 _ => {
-                    SCLogDebug!("Invalid state for request {}", self.state);
-                    current = b"";
+                    // We have gotten out of sync with the expected state flow.
```

Comment:
How is this reachable ?
Which value from the rust enum is missing ?

Reply:
We currently have a global state (i.e. that is used for both directions), but if traffic can not be parsed as expected, we might not have advanced a state for one direction but received data in the "unexpected" direction, causing the parser to end up here. In that case I decided to stop parsing further.

Reply:
> How is this reachable ?
> Which value from the rust enum is missing ?

For example, any `TC*` (to-client) state that we might still be in globally, but are calling this function which handles request (to-server) traffic.

Reply:
Thanks, maybe add a comment then ;-)

Reply:
Done.